### ISSUE TITLE
Digest::Base Error

### ISSUE THREAD

**Issue by youreddy** [fog/fog-aws#261]
Hello, 

I'm getting a `Digest::Base cannot be directly inherited in Ruby` when using this the fog-aws storage. Here's the stack trace from the error: 

```
  "Request failed: 500: {\"code\"=>10001, \"description\"=>\"Digest::Base cannot be directly inherited in Ruby\", \"backtrace\"=>[
  \"vendor/bundle/ruby/2.3.0/gems/fog-aws-0.9.1/lib/fog/aws/storage.rb:588:in `digest'\", 
  \"vendor/bundle/ruby/2.3.0/gems/fog-aws-0.9.1/lib/fog/aws/storage.rb:588:in `hexdigest'\", 
  \"vendor/bundle/ruby/2.3.0/gems/fog-aws-0.9.1/lib/fog/aws/storage.rb:588:in `request'\",
   \"vendor/bundle/ruby/2.3.0/gems/fog-aws-0.9.1/lib/fog/aws/requests/storage/get_bucket.rb:43:in `get_bucket'\",
   \"vendor/bundle/ruby/2.3.0/gems/fog-aws-0.9.1/lib/fog/aws/models/storage/directories.rb:21:in `get'\",
```

From a quick google search, it looks like this is a reoccurring issue in Digest that others have solved used `OpenSSL::Digest` instead. 

Here are some links to related issues that might be helpful: 
https://github.com/aws/aws-sdk-ruby/pull/663
https://github.com/chef/chef/pull/1807

**Comment by geemus** [comment]
Thanks for the heads up, would you be up for helping on a pull request? Thanks!

**Comment by geemus** [comment]
Closed by #266
